server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailSpooler.java

```diff
@@ -105,9 +105,8 @@ public void init() {
 
     private void run() {
         LOGGER.info("Queue={}", queue);
-
         disposable = Flux.from(queue.deQueue())
-            .flatMap(item -> handleOnQueueItem(item).subscribeOn(spooler))
+            .flatMap(item -> handleOnQueueItem(item).subscribeOn(spooler), numThreads)
```

Comment:
why do you need to specify the number of thread when we have a bounded ThreadPool?
If we have to, then shouldn't we restrict this number to `numThread - 2` to let some room for other uses of this pool?

Reply:
because by default the parallelism level of the flatMap method is equal to 
`Integer.parseInt(System.getProperty("reactor.bufferSize.x", "32")))`
So for numthread greeter than 32 it will make a difference.